### PR TITLE
Create runtimecore branch off tag 8.1

### PR DESCRIPTION
**Pull request by bweather**
Issue: https://devtopia.esri.com/runtime/milano/issues/483

Supercedes PR: https://github.com/Esri/PROJ/pull/2

https://runtime-rtc.esri.com/view/vTest/job/vtest/job/dev/job/3rdparty-interface/50/downstreambuildview/
https://runtime-rtc.esri.com/view/vTest/job/vtest/job/dev/job/3rdparty-interface/51/downstreambuildview/
https://runtime-rtc.esri.com/view/vTest/job/vtest/job/dev/job/3rdparty-interface/52/downstreambuildview/
https://runtime-rtc.esri.com/view/vTest/job/vtest/job/dev/job/3rdparty-interface/53/downstreambuildview/

### CODE REVIEW

File: PROJ.lua [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+files {
+  "src/*.h",
+  "src/*.c",
+  "src/*.hpp",
+  "src/*.cpp",
+}
+
+excludes {
+   "src/apps/**",
```
Comment (chri7325) [inline PROJ.lua]: nitpick spacing is off here

File: PROJ.lua [hunk 2]
```diff
@@ -0,0 +1,51 @@
+project "PROJ"
+
+dofile(_BUILD_DIR .. "/static_library.lua")
+
+configuration { "*" }
+
+uuid "E63D9AD5-9F32-4772-B710-38EFF074067C"
+
+defines {
```
Comment (chri7325) [inline PROJ.lua]: nitpick: Not needed

File: PROJ.lua [hunk 3]
```diff
@@ -0,0 +1,51 @@
+project "PROJ"
+
+dofile(_BUILD_DIR .. "/static_library.lua")
+
+configuration { "*" }
+
+uuid "E63D9AD5-9F32-4772-B710-38EFF074067C"
```
Comment (chri7325) [inline PROJ.lua]: Jut double checking that this is unique and wasn't copy pasted?

File: PROJ.lua [hunk 4]
```diff
@@ -0,0 +1,51 @@
+project "PROJ"
```
Comment (chri7325) [inline PROJ.lua]: This should be all lowercase as proj

File: conanfile_rtc.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    # RTC specific triple
+    settings = "platform_architecture_target"
+
+    def package(self):
+        base = self.source_folder + "src/"
+        relative = "3rdparty/PROJ/"
+
+        # headers
+        self.copy("*.h*", src=base, dst=relative)
```
Comment (chri7325) [inline conanfile_rtc.py]: Are excludes needed here too like the lua file has?

File: src/proj_config.h [hunk 6]
```diff
@@ -0,0 +1,382 @@
+#ifndef PROJ_CONFIG_H
+#define PROJ_CONFIG_H
+
+#ifdef RTC_COCOA_FAMILY
+
+/* src/proj_config.h.  Generated from proj_config.h.in by configure.  */
+/* src/proj_config.h.in.  Generated from configure.ac by autoheader.  */
+
+/* define if the compiler supports basic C++11 syntax */
+#define HAVE_CXX11 1
```
Comment (chri7325) [inline src/proj_config.h:10]: Skimming this file, I don't immediately see any differences in config from platform to platform. Do we even need these blocks after all?

**Review by chri7325 — COMMENTED** [review]
Minor nitpicks. Looks great overall. Very glad no direct changes to source is needed for trivial upgrades.

File: PROJ.lua (hunk 3 above)
Comment (bweather) [reply PROJ.lua]: I regenerated it just to be sure.

File: src/proj_config.h (hunk 6 above)
Comment (bweather) [reply src/proj_config.h:10]: ~~I believe it builds with an empty `proj_config.h`~~

File: conanfile_rtc.py (hunk 5 above)
Comment (bweather) [reply conanfile_rtc.py]: Yes, will fix.

File: src/proj_config.h (hunk 6 above)
Comment (chri7325) [reply src/proj_config.h:10]: Apologies, I meant the RTC_ ifdefs. If they're all the same we can probably just remove the need to have them separated.

File: conanfile_rtc.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+from conans import ConanFile
+
+
+class PROJConan(ConanFile):
+    name = "PROJ"
```
Comment (bweather) [inline conanfile_rtc.py:5]: @chri7325 Does this need to be lower case?

Comment (chri7325) [reply conanfile_rtc.py:5]: @bweather not the package name. Lowercase is only used in the project so the library name is lowercase to match the coding standards.

**Comment by bweather** [comment]
> base already has src?

The main header `proj.h` is contained in the `src` directory.

File: conanfile_rtc.py [hunk 8]
```diff
@@ -16,7 +16,7 @@ def package(self):
         relative = "3rdparty/PROJ/"
 
         # headers
-        self.copy("proj.h", src=base + "src", dst=relative)
+        self.copy("src/proj.h", src=base + "src", dst=relative)
```
Comment (chri7325) [inline conanfile_rtc.py]: The base variable already adds `src` so adding it again would give you `src=${path} + "src".  Can you test this locally first as I think the header is not going to go where you expect it to.

**Review by chri7325 — APPROVED** [review]

**Pull request merged**